### PR TITLE
py-exarl: new package

### PR DESCRIPTION
**Pull request by eugeneswalker**
Adding a new package for `py-exarl`

Package from 
* https://github.com/exalearn/EXARL

FYI @wspear

### CODE REVIEW

**Comment by alecbcs** [comment]
@eugeneswalker I think you might have accidentally based this off of #35823 instead of develop 😅.

**Comment by eugeneswalker** [comment]
> @eugeneswalker I think you might have accidentally based this off of #35823 instead of develop 😅.

Yes, you are right! Should be fixed now. Nice, quick catch on that. Thank you.

File: var/spack/repos/builtin/packages/py-exarl/package.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/py-exarl/package.py]: @eugeneswalker I believe the issue with the style is that the last header date should read `2023` not 2021.

Comment (eugeneswalker) [reply var/spack/repos/builtin/packages/py-exarl/package.py]: Just pushed cd4a4c28ecdebfaaa26eb1e0f9d88f03daad4551

Comment (alecbcs) [reply var/spack/repos/builtin/packages/py-exarl/package.py]: Sweet! Looks good to me!

**Comment by eugeneswalker** [comment]
Converting to draft pending feedback from `exarl` development team.

**Review by wspear — APPROVED** [review]

**Pull request merged**